### DISCUSSION TITLE
Bulk insert POINT

### DISCUSSION THREAD

**Discussion by pihentagy** [sidorares/node-mysql2#1553] [Q&A] [answered]
I have found https://github.com/sidorares/node-mysql2/issues/234 but I still cannot get mysql2 to BULK INSERT handle POINTs.

Till now I have an sql like this:
```ts
const sql = `INSERT INTO addresses (
    eov,
    settlement,
    street_name,
    created_at,
    updated_at)
VALUES ?`
```

And I can bulk insert rows like this:
```ts
await promisePool.query(sql, [rows]);
```
where every row is an array, like
```ts
[{"x":165830,"y":6809491},"Ágasegyháza","Ady Endre","2022-04-04T16:26:27.606Z","2022-04-04T16:26:27.606Z"]
```
Except that the eov field which is a new field with type POINT is not filled.

```
Error: Cannot get geometry object from data you send to the GEOMETRY field
```

Tried with the `{ type: 'Point', coordinates: [39.807222,-76.984722] }` format, but neither formats worked.

**Comment by sidorares** [comment, marked as answer]
this example should help: https://github.com/sidorares/node-mysql2/issues/1244#issuecomment-1094547067

**Reply by pihentagy** [reply]
Thanks, but just going FROM that direction (csv file on mysql server). Sending tabular data directly to mysql is error prone (it eats everything, including textual data on numbers), so to improve error handling I read the file via a csv parser.

So I guess I cannot avoid creating a real object for every point, right?

**Reply by sidorares** [reply]
I need to investigate if its possible to serialize POINT as one single parameter ( same way it's serialized back from server ), so potentially when this is implemented you should be able to do one placeholder per point with `.execute()` but in general you probably can't avoid creating a real object for every point.
